From 626ba6d88fd1a8076062c91645c1db1175f4962f Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Thu, 27 Dec 2018 09:07:48 +0800
Subject: [PATCH] media: rockchip: isp1: to speed up stream off

current frame end to stop mp or sp, and next
frame to close, will take a frame time.
change to stop immediately and current frame end
to close.

Change-Id: I7d1b8fa23d13de1ee637d076ee0e75cbcb62e7e6
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rockchip/isp1/capture.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/media/platform/rockchip/isp1/capture.c b/drivers/media/platform/rockchip/isp1/capture.c
index 7cfc52946661..6a0cc8ba3c19 100644
--- a/drivers/media/platform/rockchip/isp1/capture.c
+++ b/drivers/media/platform/rockchip/isp1/capture.c
@@ -1232,12 +1232,12 @@ static void rkisp1_stream_stop(struct rkisp1_stream *stream)
 	int ret = 0;
 
 	stream->stopping = true;
+	stream->ops->stop_mi(stream);
 	ret = wait_event_timeout(stream->done,
 				 !stream->streaming,
 				 msecs_to_jiffies(1000));
 	if (!ret) {
 		v4l2_warn(v4l2_dev, "waiting on event return error %d\n", ret);
-		stream->ops->stop_mi(stream);
 		stream->stopping = false;
 		stream->streaming = false;
 	}
@@ -2176,8 +2176,6 @@ void rkisp1_mi_isr(u32 mis_val, struct rkisp1_device *dev)
 				stream->stopping = false;
 				stream->streaming = false;
 				wake_up(&stream->done);
-			} else {
-				stream->ops->stop_mi(stream);
 			}
 		} else {
 			mi_frame_end(stream);
-- 
2.35.3

